### PR TITLE
Fix reports with `user == undefined`

### PR DESCRIPTION
**Pull request by pdcp1**
This PR fixes https://github.com/responsible-ai-collaborative/aiid/issues/2237

- Fix the current reports with `user == undefied`
- Set `null` `user` value for submissions that are submitted anonymously

### CODE REVIEW

**Review by cesarvarela — COMMENTED** [review]
Looks good, but I'm unsure about setting Sean up as de default submitter.

@smcgregor @kepae ?

File: site/gatsby-site/migrations/2023.08.17T15.56.33.fix-report-users.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  const updateResult = await reportsCollection.updateMany(
+    {
+      user: undefined,
+    },
+    { $unset: { user: '' } }
+  );
+
+  console.log('Update result:', updateResult);
+};
```
Comment (kepae) [inline site/gatsby-site/migrations/2023.08.17T15.56.33.fix-report-users.js:21]: Please consider adding the empty `down` function even though there's no logical way to revert -- this way the node migrator tool won't throw an error on down migrations.

**Comment by kepae** [comment]
Agree. It may be better to create a default editor user (with no real permissions) that is only used for backfilling submission metadata.

Would it be possible to create a default editor user?

File: site/gatsby-site/migrations/2023.08.17T15.56.33.fix-report-users.js (hunk 1 above)
Comment (pdcp1) [reply site/gatsby-site/migrations/2023.08.17T15.56.33.fix-report-users.js:21]: added

**Comment by pdcp1** [comment]
We are currently usign Sean as a default Editor when promoting a submission (see `promoteSubmissionToReport.js` code below), so I keep consistency for the `user` as well.

```
const editors = (!submission.incident_editors || !submission.incident_editors.length)
        ? [SEAN_USER_ID]
        : submission.incident_editors;
```

**Comment by kepae** [comment]
Ah sounds good, thanks for explaining -- it's the "existing" logic. We shouldn't block fixing the bug on it, so LGTM.

**Comment by smcgregor** [comment]
+1 with having a system default user.

**Comment by kepae** [comment]
Opened https://github.com/responsible-ai-collaborative/aiid/issues/2240 to track that. Since this bugfix doesn't change existing behavior, I don't believe we should block it in this PR -- unless it would explode the number of submissions attached to your user, but I don't think it should if this only affects old submissions that pre-dated user logic.

**Comment by kepae** [comment]
Ah, wait. editor != user.

Will submissions from anonymous users now be attributed to Sean?

**Comment by pdcp1** [comment]
After further analysis, we decided to set the `report.user = null` (instead of Sean user) for submissions that are submitted by an anonymous user.
Currently, we have more than 2500 reports in this situation and the report edition works fine.

In the future, when we have a Default user, we can assign this Default user when an anonymous user submit a submission.

**Review by cesarvarela — COMMENTED** [review]
LGTM I'll wait for the deploy to finish before merging

**Review by kepae — APPROVED** [review]

**Pull request merged**